app/models/concerns/has_season.rb

```diff
@@ -4,5 +4,12 @@ module HasSeason
 
   included do
     belongs_to :season
+
+    # @param season [#name, String, Integer]
+    # @return [ActiveRecord::Relation]
+    def self.in_season(season)
+      season = season.try(:name) || season
```

Comment:
Personal preference, but I'd find
```ruby
name = season&.name || season
```
easier on the eyes... Plus: I think it's good practice to stop using `try` since it's basically in the language now 😉

Reply:
The argument "union type" requires the `try`. There is a difference in `try` and `&`, they're not the same thing:

Object#try is
```ruby
def try(meth)
  send(meth) if respond_to?(meth)
end
```

The safe navigation is:
```
send(meth) unless nil?
```

```
$ rails c
Loading development environment (Rails 5.1.4)
2.5.0 :001 > RUBY_VERSION
 => "2.5.0"
2.5.0 :002 > "".try(:name)
 => nil
2.5.0 :003 > nil.try(:name)
 => nil
2.5.0 :004 > nil&.name
 => nil
2.5.0 :005 > ""&.name
Traceback (most recent call last):
NoMethodError (undefined method `name' for "":String)
```

Reply:
🙃  I didn't realize this. Thnks

Reply:
Oh, did not think this through 🙈 thanx for pointing out @carpodaster 🙏

Reply:
Whoa, thanks! Didn't know this 👍